src/neo/SmartContract/InteropService.cs

```diff
@@ -520,19 +520,46 @@ private static bool PutEx(ApplicationEngine engine, StorageContext context, byte
                 Key = key
             };
 
-            if (engine.Snapshot.Storages.TryGet(skey)?.IsConstant == true) return false;
+            var skeyValue = engine.Snapshot.Storages.TryGet(skey);
+            if (skeyValue?.IsConstant == true) return false;
+
+            var currentOccupiedBytes = skeyValue == null ? 0 : skeyValue.Size;
+            var releasedBytes = 0;
+            var reusedBytes = 0;
 
             if (value.Length == 0 && !flags.HasFlag(StorageFlags.Constant))
             {
                 // If put 'value' is empty (and non-const), we remove it (implicit `Storage.Delete`)
                 engine.Snapshot.Storages.Delete(skey);
+                releasedBytes = currentOccupiedBytes;
             }
             else
             {
                 StorageItem item = engine.Snapshot.Storages.GetAndChange(skey, () => new StorageItem());
                 item.Value = value;
                 item.IsConstant = flags.HasFlag(StorageFlags.Constant);
+                if (value.Length < currentOccupiedBytes)
+                {
+                    releasedBytes = currentOccupiedBytes - value.Length;
+                    reusedBytes = currentOccupiedBytes - releasedBytes;
+                }
+                else
+                {
+                    reusedBytes = currentOccupiedBytes;
+                }
+
```

Comment:
Better to remove this blank line

Reply:
Shouldn't dotnet format take care of this? I will remove and see if it complains.

Reply:
Fixed. I will be more cautious next time.